[docs] Move the demo higher in the API TOC

oliviertassinari opened the pull request:
It might solve these feedback:

<img width="336" alt="Screenshot 2022-11-19 at 01 35 45" src="https://user-images.githubusercontent.com/3165635/202824698-bc2dc92d-c10c-4269-9efc-9bbd7067f12f.png">

https://mui-org.slack.com/archives/C041SDSF32L/p1668804057511509

When I'm landing on the API page, I usually have to scroll to the bottom of the page to then go to the demo (it happens to me quite often). Maybe this is a better UX https://deploy-preview-35202--material-ui.netlify.app/material-ui/api/alert/. I have made sure that searching for "example" yield a result too on the page.

Close #35004

mnajdova reviewed (approved):
👍

alexfauquette commented:
One of the ideas we had when discussing the issue was to add an explanation. Hopping it helps people who land on this page discover the structure of the docs. Especially the "Components" section.

Here is the text proposed in the issue:

> This API page is a quick reference for the props and classes available to this component. To see demos and details on how to use the component in your app, visit the corresponding page in the **Components** section of the docs:

oliviertassinari commented:
@alexfauquette Ok, I have updated the description.

- I have skipped "This API page is a quick reference for the props and classes available to this component." because it says the same as the description:

<img width="471" alt="Screenshot 2022-11-24 at 23 09 52" src="https://user-images.githubusercontent.com/3165635/203869007-baab99b6-f0b8-4beb-8282-912e0fb75cbf.png">

- I have shortened it to stay on one line.
- I have replaced "details" with "example" so developers can keyword search on the page.

samuelsycamore commented on docs/src/modules/components/ApiPage.js:
```diff
@@ -298,6 +298,14 @@ function ApiDocs(props) {
+        <p
+          dangerouslySetInnerHTML={{
+            __html:
+              'To see demos and examples on how to use this React component, visit the component guide section:',
```
Here's how I would phrase this:

```suggestion
              'For examples and details on the usage of this React component, visit the component demo page:',
```

samuelsycamore replied:
Another thought—to teach users where to find the demos in the side nav:

```suggestion
              'For examples and details on the usage of this React component, visit the corresponding <b>Component</b> page:',
```

oliviertassinari replied:
The second proposal doesn't work with MUI X and makes the line wrap. I have used the first one.

samuelsycamore commented:
Would it make more sense for this section to be a callout? It feels kind of weird to be the first section of the page, but not really a part of the page's flow. I think it should be somewhat eye-catching, since the purpose is to quickly redirect users who've landed on the wrong page.

oliviertassinari commented:
> Would it make more sense for this section to be a callout?

@samuelsycamore A callout would make the information more likely to be seen by a developer. IMHO, this would be too much, too soon. We already give it a lot more visibility by moving it at the top.
